src/main/java/org/folio/rest/impl/HoldingsRecordsSourceAPI.java

```diff
@@ -0,0 +1,193 @@
+                msg =  (msg == null) ? "Internal server problem: Error message missing" : msg;
+                log.info(msg);
+                asyncResultHandler.handle(succeededFuture(DeleteHoldingsRecordsSourcesByIdResponse
+                  .respond400WithTextPlain((msg))));
+                return;
+              }
+              int updated = reply.result().rowCount();
+              if (updated != 1) {
+                String msg = messages.getMessage(lang, MessageConsts.DeletedCountError, 1, updated);
+                log.error(msg);
+                asyncResultHandler.handle(succeededFuture(DeleteHoldingsRecordsSourcesByIdResponse
+                  .respond404WithTextPlain(msg)));
+                return;
+              }
+              asyncResultHandler.handle(succeededFuture(DeleteHoldingsRecordsSourcesByIdResponse
+                .respond204()));
+            });
+        } catch (Exception e) {
+          log.error(e.getMessage(), e);
+          asyncResultHandler.handle(succeededFuture(DeleteHoldingsRecordsSourcesByIdResponse
+            .respond500WithTextPlain(messages.getMessage(lang, MessageConsts.InternalServerError))));
+        }
+      });
+  }
+
+  @Override
+  public void putHoldingsRecordsSourcesById(String id, String lang,
+    HoldingsRecordsSource entity, Map<String, String> okapiHeaders,
+    Handler<AsyncResult<Response>> asyncResultHandler, Context vertxContext) {
+      vertxContext.runOnContext(v -> {
+        String tenantId = TenantTool.tenantId(okapiHeaders);
+        try {
+          if (entity.getId() == null) {
+            entity.setId(id);
+          }
+          PostgresClient.getInstance(vertxContext.owner(), tenantId).update(REFERENCE_TABLE, entity, id,
+            reply -> {
+              if (reply.succeeded()) {
+                if (reply.result().rowCount() == 0) {
+                  asyncResultHandler.handle(succeededFuture(PutHoldingsRecordsSourcesByIdResponse
+                      .respond404WithTextPlain(messages.getMessage(lang, MessageConsts.NoRecordsUpdated))));
+                } else{
+                  asyncResultHandler.handle(succeededFuture(PutHoldingsRecordsSourcesByIdResponse
+                      .respond204()));
+                }
+              } else {
+                String msg = PgExceptionUtil.badRequestMessage(reply.cause());
+                msg =  (msg == null) ? "Internal server problem: Error message missing" : msg;
+                log.info(msg);
+                asyncResultHandler.handle(succeededFuture(PutHoldingsRecordsSourcesByIdResponse
+                    .respond400WithTextPlain(msg)));
+              }
+            });
+        } catch (Exception e) {
+          log.error(e.getMessage(), e);
+          asyncResultHandler.handle(succeededFuture(PutHoldingsRecordsSourcesByIdResponse
+            .respond500WithTextPlain(messages.getMessage(lang, MessageConsts.InternalServerError))));
+        }
+      });
```

Comment:
Is there any reason why standard org.folio.rest.persist.PgUtil#put(...) method can not be used?